rpcserverhandlers_test.go

```diff
@@ -3450,9 +3585,30 @@ func testRPCServerHandler(t *testing.T, tests []rpcTest) {
 			}
 			if err != nil {
 				t.Errorf("%s\nunexpected error: %+v\n", test.name, err)
+				return
 			}
-			if !reflect.DeepEqual(result, test.result) {
-				t.Errorf("%s\nwant: %+v\n got: %+v\n", test.name, test.result, result)
+			// Tests that return pointer values are unwrapped per case.
+			switch want := test.result.(type) {
+			case *types.EstimateStakeDiffResult:
+				got := result.(*types.EstimateStakeDiffResult)
+				var wantUser, gotUser float64
+				if want.User != nil {
+					wantUser = *want.User
+				}
+				if got.User != nil {
+					gotUser = *got.User
+				}
+				if want.Min != got.Min ||
+					want.Max != got.Max ||
+					want.Expected != got.Expected ||
+					wantUser != gotUser {
+					t.Errorf("%s\nwant: %+v with user value %v\n got: %+v with user value %v\n",
+						test.name, want, wantUser, got, gotUser)
+				}
+			default:
+				if !reflect.DeepEqual(result, test.result) {
+					t.Errorf("%s\nwant: %+v\n got: %+v\n", test.name, test.result, result)
```

Comment:
I'm not sure I follow, although printing structs after failing with go-spew is a great idea, the special case is because `reflect.DeepEqual()` will compare pointer values and not the underlying `User` value the pointer points to. Unless I am mistaken.

Reply:
Actually should work, in docs `Pointer values are deeply equal if they are equal using Go's == operator or if they point to deeply equal values.` So maybe I was doing something weird, let me try again...

Reply:
Actually, working now. I must have been doing something wierd. Apologies. But go-spew will be an asset.